### PR TITLE
feat: view recording button in hogql

### PR DESCRIPTION
**Pull request by pauldambra**
image

@camerondeleone was asking about how to link to recordings from a hogql insight...

but really we want to play them...

let's make that possible by adding a new function and renderer to hogql

NB this won't work in shared insights/dashboards, since we're not sharing _every_ recording just in case it is in the insight's results

### CODE REVIEW

**Review by mariusandra — APPROVED** [review]
This feels like just the right amount of bonkers.

File: frontend/src/queries/nodes/HogQLX/render.tsx [hunk 1]
```diff
@@ -38,6 +43,30 @@ export function renderHogQLX(value: any): JSX.Element {
...
+                        onClick={(e) => {
+                            e.preventDefault()
+                            if (sessionId) {
+                                openSessionPlayer({ id: sessionId })
+                            }
+                        }}
+                    >
+                        View recording
+                    </LemonButton>
+                    <SessionPlayerModal />
```
Comment (daibhin) [inline frontend/src/queries/nodes/HogQLX/render.tsx:67]: Will this not add a modal for each rendered row? Maybe we could be clever and put this component in the header of the table to make sure it's rendered

File: frontend/src/scenes/session-recordings/player/modal/SessionPlayerModal.tsx [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+/**
+ * When SessionPlayerModal is present in the page you can call `openSessionPlayer` action to open the modal
+ * and play a given session
+ *
+ * SessionPlayerModal is templated into the page by multiple components
+ * It has to be present _once_ for the player modal to work
+ * But gets very unhappy if there are multiple instances
+ * So, it is written as a singleton that attempts to only render once
+ *
+ */
```
Comment (daibhin) [inline frontend/src/scenes/session-recordings/player/modal/SessionPlayerModal.tsx:21]: Please ignore my above comment 😂 I should have read the entire PR first

Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/modal/SessionPlayerModal.tsx:21]: no, you're right, it's equally messy now :)

**Comment by daibhin** [comment]
We were hesitant to add HogQLX because the use cases weren't immediately obvious. `Sparkline` has been useful and I think this is a really good second!

**Comment by mariusandra** [comment]
The biggest real (yet presently hidden) usecase for HogQLx is the `<a href={properties.$current_url}>{event}</a>` tag. You can make your own links!

**Comment by kektobiologist** [comment]
umm noob question but how do i use this? I'm using a query to fetch relevant session_ids like so
```
SELECT substring(message, 1, 50) as message_prefix, count(*) as cnt, first_value(timestamp), first_value(message), first_value(log_source_id) as session_id
FROM log_entries 
where log_source = 'session_replay' and level = 'warn' 
and message not like '"[SessionRecording]%'
and message not like '"[Intercom]%'
and message not like '"Error parsing SQL%'
and dateDiff('hour', timestamp, now()) < 36
group by message_prefix having cnt > 20
order by cnt desc
LIMIT 500
```
is there an easy way for me to view the recordings of the resulting session_ids? in fact i'm not sure how to manually get the recordings for any particular session id either (couldn't find any docs for which table it is in; couldn't figure out how to filter by session_id in the session replay tab either)

**Comment by pauldambra** [comment]
@kektobiologist 

> SELECT substring(message, 1, 50) as message_prefix, count(*) as cnt, first_value(timestamp), first_value(message), first_value(log_source_id) as session_id
FROM log_entries 

if you add `recording_button(first_value(log_source_id))` as a column that should work.

I didn't check the query but the way to get the recording button is to add `recording_button($the_session_id)`

> couldn't figure out how to filter by session_id in the session replay tab

the easiest way to view a single recording is to visit `https://us.posthog.com/replay/$the_session_id`

or to hit CMD + SHIFT + K and paste the session id in there

**Comment by kektobiologist** [comment]
thanks, all of this worked 🙏

**Pull request merged**